From f642509c7c83d5dd4652eaa6975162437bc3627d Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Sat, 12 Oct 2024 15:30:59 -0400
Subject: [PATCH 23/25] init: add VENDOR_THERMAL_ENABLE boot flag

This will give us a way to disable the Intel thermal HAL on devices.
---
 init.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/init.sh b/init.sh
index 1ca9d2a..ef2bc82 100644
--- a/init.sh
+++ b/init.sh
@@ -570,10 +570,14 @@ function init_hal_thermal()
 	#thermal-daemon test, pulled from Project Celadon
 	case "$(cat /sys/class/dmi/id/chassis_vendor | head -1)" in 
 	QEMU)
-		setprop vendor.thermal.enable 0
+		setprop vendor.thermal.enable ${VENDOR_THERMAL_ENABLE:-0}
 		;;
 	*)
-		setprop vendor.thermal.enable 1
+		if [ $(cat /proc/cpuinfo | grep vendor_id | tr ':' ' ' | cut -d' ' -f 3| head -1) == "GenuineIntel" ]; then
+			setprop vendor.thermal.enable ${VENDOR_THERMAL_ENABLE:-1}
+		else
+			setprop vendor.thermal.enable ${VENDOR_THERMAL_ENABLE:-0}
+		fi
 		;;
 	esac
 }
-- 
2.34.1

